gneiss/stats/composition.py

```diff
@@ -0,0 +1,43 @@
+import numpy as np
+from skbio import DistanceMatrix
+from skbio.stats.composition import closure
+
+
+def variation_matrix(X):
+    """ Calculate Aitchison variation matrix.
+
+    This calculates the Aitchison variation matrix.  Given a compositional
+    matrix :math:`X`, row :math:`i` and column:math:`j`, in the variation
+    matrix of :math:`X` is given by
+
+    .. math:
+        V_{ij} = \frac{1}{2} var(\ln \frac{x_i}{x_j})
+
+    Parameters
+    ----------
+    X : pd.DataFrame
+        Contingency table where there are n rows corresponding to samples
+        and p features corresponding to columns.
+
+    Returns
+    -------
+    skbio.DistanceMatrix
+        Total variation matrix of size n x n.
+
+    References
+    ----------
+    .. [1] V. Pawlowsky-Glahn, J. J. Egozcue, R. Tolosana-Delgado (2015),
+       Modeling and Analysis of Compositional Data, Wiley, Chichester, UK
+
+    .. [2] J. J. Egozcue, V. Pawlowsky-Glahn (2004), Groups of Parts and
+       Their Balances in Compositional Data Analysis, Mathematical Geology
+    """
+    v = np.zeros((X.shape[1], X.shape[1]))
+    x = closure(X)
+    for i in range(X.shape[1]):
+        for j in range(i):
+            v[i, j] = np.var(np.log(x[:, i]) - np.log(x[:, j]))
```

Comment:
why not `np.log(x[j, :])` if description above is true about comparing one sample with one feature?